river/metrics/base.py

```diff
@@ -84,7 +84,7 @@ def bigger_is_better(self):
         return True
 
     def works_with(self, model) -> bool:
-        return utils.inspect.isclassifier(model)
+        return utils.inspect.isclassifier(model) or utils.inspect.isclusterer(model)
```

Comment:
I have thought about it when I updated this line. However, this is also the reason why I put `or` instead of `and`. This will also facilitate us in the evaluation file later on. @jacobmontiel Do you have any other thoughts?

Reply:
Although this still needs to be discussed, functionality-wise I am pretty happy with the results.

Hence, I'll mark this PR as approved, from my side!

Reply:
We can override the `works_with` method in `MultiClassMetric` so binary metrics will only work with classifiers and multi-class metrics will work with both classifiers and clusteres.